addons/product/models/product_pricelist.py

```diff
@@ -310,34 +310,49 @@ def _get_partner_pricelist(self, partner_id, company_id=None):
         res = self._get_partner_pricelist_multi([partner_id], company_id)
         return res[partner_id].id
 
+    def _get_partner_pricelist_multi_search_domain_hook(self):
+        return []
+
+    def _get_partner_pricelist_multi_filter_hook(self):
+        return self
+
     def _get_partner_pricelist_multi(self, partner_ids, company_id=None):
         """ Retrieve the applicable pricelist for given partners in a given company.
 
+            It will return the first found pricelist in this order:
+            First, the pricelist of the specific property (res_id set), this one
+                   is created when saving a pricelist on the partner form view.
+            Else, it will return the pricelist of the partner country group
+            Else, it will return the generic property (res_id not set), this one
+                  is created on the company creation.
+            Else, it will return the first available pricelist
+
             :param company_id: if passed, used for looking up properties,
                 instead of current user's company
             :return: a dict {partner_id: pricelist}
         """
-        Partner = self.env['res.partner']
+        Partner = self.env['res.partner'].with_context(active_test=False)
```

Comment:
comment that you need inactive for public user

is it true ? public user use property too?

Reply:
Yes of course (from what I understood from current code implementation).
The website public user is inactive.
Still, we need to retrieve its `property_product_pricelist` as [it will be passed](https://github.com/odoo-dev/odoo/blame/905a4a4be435011668f6994bb2d0e4d16ff2e5aa/addons/website_sale/models/website.py#L127) to the `_get_pl_partner_order()` method as the [default website pricelist](https://github.com/odoo-dev/odoo/blame/905a4a4be435011668f6994bb2d0e4d16ff2e5aa/addons/website_sale/models/website.py#L55) (`website_pl`) parameter.
I already wrote a test for that and explained why, see [this test](https://github.com/odoo-dev/odoo/blame/905a4a4be435011668f6994bb2d0e4d16ff2e5aa/addons/website_sale/tests/test_website_sale_pricelist.py#L232).

Reply:
As discussed, it was really a unwanted/unnoticed side effect of https://github.com/odoo-dev/odoo/commit/d39d583b213af02bad12721d9c610def6cb09161
It would do
```python
p = Partner.browse(partner_id)
```
And now it will do
```python
for pid in Partner.search(group['__domain']).ids:
```
Browse would return inactive while search dont.
See last message for explanation about website public user:

> Still, we need to retrieve its property_product_pricelist as it will be passed to the _get_pl_partner_order() method as the default website pricelist (website_pl) parameter.
> I already wrote a test for that and explained why, see this test.